Replace boost with standard library equivalents

Dr15Jones opened the pull request.

Dr15Jones committed:
Replace boost with std equivalent

cmsbuild commented:
A new Pull Request was created by  @Dr15Jones (Chris Jones) for master.

It involves the following packages:

DataFormats/Common
DataFormats/FWLite
DataFormats/Provenance
FWCore/Framework
FWCore/ParameterSet
FWCore/ServiceRegistry
FWCore/Utilities


@cmsbuild, @smuzaffar, @Dr15Jones, @davidlange6 can you please review it and eventually sign? Thanks.
@makortel, @felicepantaleo, @ebrondol, @wddgit, @Martin-Grunewald this is something you requested to watch as well.
@davidlange6 you are the release manager for this.

cms-bot commands are listed <a href="http://cms-sw.github.io/cms-bot-cmssw-cmds.html">here</a>

Dr15Jones commented:
please test

cmsbuild commented:
The tests are being triggered in jenkins.

Dr15Jones commented:
+1

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
-1

Tested at: b8f2145242336a07aed0b63d759ad4a8024de767


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/13dfecf9718a57323e839740705e4b7af07742ff
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21522/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21522/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21522/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>Selected class -> std::map<int, std::vector<CSCNoiseMatrix::Item, std::allocator<CSCNoiseMatrix::Item> >, std::less<int>, std::allocator<std::pair<const int, std::vector<CSCNoiseMatrix::Item, std::allocator<CSCNoiseMatrix::Item> > > > > for ROOT: map<int,vector<CSCNoiseMatrix::Item> >
Selected class -> std::map<int, std::vector<CSCcrosstalk::Item, std::allocator<CSCcrosstalk::Item> >, std::less<int>, std::allocator<std::pair<const int, std::vector<CSCcrosstalk::Item, >> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-16-2300/src/Geometry/CSCGeometry/src/CSCChamber.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-16-2300/src/Geometry/CSCGeometry/src/CSCLayerGeometry.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-16-2300/src/Geometry/CSCGeometry/src/CSCGattiFunction.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-16-2300/src/Geometry/CSCGeometry/src/CSCGangedWireGrouping.cc: In member function 'virtual int CSCGangedWireGrouping::wireGroup(int) const':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-16-2300/src/Geometry/CSCGeometry/src/CSCGangedWireGrouping.cc:81:3: error: 'ptrdiff_t' was not declared in this scope
   ptrdiff_t pd = it - theFirstWireOfEachWireGroup.begin();
   ^~~~~~~~~
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-16-2300/src/Geometry/CSCGeometry/src/CSCGangedWireGrouping.cc:81:3: note: suggested alternatives:
In file included from /cvmfs/cms-ib.cern.ch/nweek-02481/slc6_amd64_gcc630/external/gcc/6.3.0/include/c++/6.3.0/bits/stl_algobase.h:59:0,
                 from /cvmfs/cms-ib.cern.ch/nweek-02481/slc6_amd64_gcc630/external/gcc/6.3.0/include/c++/6.3.0/vector:60,
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/13dfecf9718a57323e839740705e4b7af07742ff
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21522/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21522/git-merge-result

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

Dr15Jones committed:
Added missing include of cstddef

A change to the framewrok removed that header from framework code
but it is needed here.

cmsbuild commented:
Pull request #19773 was updated. @smuzaffar, @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @davidlange6 can you please check and sign again.

Dr15Jones commented:
please test

Dr15Jones commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
This pull request is fully signed and it will be integrated in one of the next master IBs after it passes the integration tests. This pull request requires discussion in the ORP meeting before it's merged. @davidlange6, @smuzaffar

cmsbuild commented:
-1

Tested at: 115d2c7f55f4b65742d522b8245727e1861dd4c9


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/7b1c63d9e3a493d2602cbc23157526aaac79017f
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21537/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21537/git-merge-result

You can see the results of the tests here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21537/summary.html

I found follow errors while testing this PR

**Failed tests:  Build**

* **Build**:

I found an error when building: 
 
 <pre>>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/CondCore/CondDB/src/GTSchema.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/CondCore/CondDB/src/Cipher.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/CondCore/CondDB/src/DecodingKey.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/CondCore/CondDB/src/RunInfoProxy.cc 
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/CondCore/CondDB/src/DecodingKey.cc: In function 'std::__cxx11::string cond::getLoginName()':
/build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/CondCore/CondDB/src/DecodingKey.cc:43:39: error: '::getuid' has not been declared
     struct passwd* userp = ::getpwuid(::getuid());
                                       ^~
gmake: *** [tmp/slc6_amd64_gcc630/src/CondCore/CondDB/src/CondCoreCondDB/DecodingKey.o] Error 1
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/CondCore/CondDB/src/IOVSchema.cc 
>> Compiling  /build/cmsbld/jenkins-workarea/workspace/ib-any-integration/CMSSW_9_3_X_2017-07-17-1100/src/CondCore/CondDB/src/Binary.cc 
</pre>


The following merge commits were also included on top of IB + this PR after doing git cms-merge-topic: 
https://github.com/cms-sw/cmssw/commit/7b1c63d9e3a493d2602cbc23157526aaac79017f
You can see more details here:
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21537/git-log-recent-commits
https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-19773/21537/git-merge-result

cmsbuild commented:
Comparison not run due to Build errors (RelVals and Igprof tests were also skipped)

Dr15Jones committed:
Add missing include unistd.h

The header was implicitly included from the framework. Since the
framework no longer needs it, it must be explicitly included.

cmsbuild commented:
Pull request #19773 was updated. @smuzaffar, @civanch, @Dr15Jones, @ianna, @mdhildreth, @cmsbuild, @ggovi, @davidlange6 can you please check and sign again.

Dr15Jones commented:
please test

Dr15Jones commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

ianna commented on DataFormats/Common/interface/ValueMap.h:
```diff
@@ -14,6 +14,7 @@
 #include <map>
 #include <iterator>
 #include <algorithm>
+#include <stddef.h>
```
@Dr15Jones - just curious why later cstddef is included and not here:
https://github.com/cms-sw/cmssw/pull/19773/files#diff-1588000c7ee3060dfd82d6c7ea0c1b4bR8

Dr15Jones replied:
When I googled where the missing header came from, my first hit was `stddef.h`. Probably using `cstddef` would have been a better choice.

Dr15Jones commented:
@smuzaffar the Jenkin's log seem to show the build timed out but our integration system doesn't seem to have noticed.

Dr15Jones committed:
Use cstddef instead of stddef.h

Dr15Jones commented:
please test

Dr15Jones commented:
+1

cmsbuild commented:
The tests are being triggered in jenkins.

cmsbuild commented:
Pull request #19773 was updated. @ggovi, @cmsbuild, @davidlange6 can you please check and sign again.

smuzaffar commented:
@Dr15Jones , I will fix the jenkins jobs to run some post build in case job failed or timed out

Dr15Jones commented:
Code builds fine this time

Dr15Jones commented:
@smuzaffar @davidlange6 the build timed out after 2h but looks like everything ran fine up till then. 

The only changes for **db** to sign is the inclusion of a missing header.

davidlange6 commented:
merge
